### PR TITLE
Add workspace upload support

### PR DESCRIPTION
**Pull request by ebraminio**

### CODE REVIEW

**Comment by ebraminio** [comment]
Dupe to #65

File: web/api.go [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
 // Version returns json encoded version details
 func (ws *webServer) Version(w http.ResponseWriter, r *http.Request) {
-	versionJSON, err := json.Marshal(ws.ds.SelfInfo())
+	selfInfo := ws.ds.SelfInfo()
+	currentHash, err := ws.ds.GetClusterVariable(currentWorkspaceHashKey)
+	if err != nil {
+		fmt.Println("Error while retrieving current workspace hash")
+	} else {
+		selfInfo.CurrentWorkspaceHash = currentHash
+	}
```
Comment (ebraminio) [inline web/api.go]: Check if is possible without this

File: dev_run.sh [hunk 2]
```diff
@@ -209,6 +209,19 @@ if [[ "${1:-}" == "clean" ]]; then
 fi
 
 
+#### upload workspace
```
Comment (ebraminio) [inline dev_run.sh]: should be tested once

File: web/api.go [hunk 3]
```diff
@@ -230,3 +244,93 @@ func (ws *webServer) DelClusterVariables(w http.ResponseWriter, r *http.Request)
...
+	extractPath := path.Join(workspaceParentPath, hash)
+	err = untar(tarPath, extractPath)
+	if err != nil {
+		http.Error(w, `{"error": "Error untaring the uploaded workspace"}`, http.StatusInternalServerError)
+		return
+	}
+
+	currentWorkspacePath := path.Join(workspaceParentPath, "current/")
+	destinedCurrentWorkspacePath := path.Join(extractPath, "workspace/")
+	err = os.Symlink(destinedCurrentWorkspacePath, currentWorkspacePath)
```
Comment (ebraminio) [inline web/api.go:275]: Remove current symlink before symlinking a new one

File: web/api.go [hunk 4]
```diff
@@ -230,3 +238,98 @@ func (ws *webServer) DelClusterVariables(w http.ResponseWriter, r *http.Request)
 
 	io.WriteString(w, `"OK"`)
 }
+
+func (ws *webServer) WorkspaceUploadHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
```
Comment (arastu) [inline web/api.go:243]: please get a lock to prevent race condition

File: web/api.go [hunk 5]
```diff
@@ -230,3 +236,65 @@ func (ws *webServer) DelClusterVariables(w http.ResponseWriter, r *http.Request)
...
+	}
+
+	extractPath := path.Join(workspaceParentPath, hash)
+	err = utils.Untar(tarPath, extractPath)
+	if err != nil {
+		http.Error(w, `{"error": "Error untaring the uploaded workspace"}`, http.StatusInternalServerError)
+		return
+	}
+
+	currentWorkspacePath := path.Join(workspaceParentPath, "current/")
```
Comment (ebraminio) [inline web/api.go:268]: `ws.ds.WorkspacePath()` should be used instead.

**Pull request merged**